### PR TITLE
UI: Part 5 - hds adoption replace <Modal>

### PR DESCRIPTION
**Pull request by hellobontempo**
Replaces confirmation modals

### CODE REVIEW

File: ui/lib/core/addon/components/confirmation-modal.hbs [hunk 1]
```diff
@@ -3,40 +3,45 @@
...
-    >
-      {{this.buttonText}}
-    </button>
-    <button type="button" class="button is-secondary" {{on "click" @onClose}} data-test-cancel-button>
-      Cancel
-    </button>
-  </footer>
-</Modal>
\ No newline at end of file
+{{#if @isActive}}
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/confirmation-modal.hbs:6]: Screenshot 2023-10-03 at 10 53 17 AM
Screenshot 2023-10-03 at 10 53 27 AM

File: ui/lib/core/addon/components/confirmation-modal.js [hunk 2]
```diff
@@ -15,38 +17,22 @@ import Component from '@glimmer/component';
...
-  get buttonClass() {
-    return this.args.buttonClass || 'is-danger';
-  }
-
-  get buttonText() {
-    return this.args.buttonText || 'Confirm';
-  }
-
-  get confirmText() {
-    return this.args.confirmText || 'Yes';
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/confirmation-modal.js:42]: Nowhere used the default arg so I removed this to simplify the component

File: ui/lib/core/addon/templates/components/replication-action-demote.hbs [hunk 3]
```diff
@@ -30,7 +30,6 @@
   @title="Demote to secondary?"
   @onClose={{action (mut this.isModalActive) false}}
   @isActive={{this.isModalActive}}
-  @buttonClass="is-primary"
```
Comment (hellobontempo) [inline ui/lib/core/addon/templates/components/replication-action-demote.hbs:33]: Screenshot 2023-10-03 at 10 56 02 AM

File: ui/lib/core/addon/components/confirmation-modal.hbs (hunk 1 above)
Comment (zofskeez) [reply ui/lib/core/addon/components/confirmation-modal.hbs:6]: Can this modal appear in both instances you have in the screenshots (with or without the header) or is the second screen what it looked like before?

Comment (zofskeez) [reply ui/lib/core/addon/components/confirmation-modal.hbs:6]: Gotcha thanks!

**Pull request merged**